test/AngularSpec.js

```diff
@@ -1708,83 +1708,78 @@ describe('angular', function() {
       dealoc(appElement);
     });
 
-    it('should bootstrap from an extension into an extension document for same-origin documents only', function() {
-      // IE does not support `document.currentScript` (nor extensions with protocol), so skip test.
-      if (msie) return;
-
-      // Extension URLs are browser-specific, so we must choose a scheme that is supported by the browser to make
-      // sure that the URL is properly parsed.
-      var extensionScheme;
-      var userAgent = window.navigator.userAgent;
-      if (/Firefox\//.test(userAgent)) {
-        extensionScheme = 'moz-extension';
-      } else if (/Edge\//.test(userAgent)) {
-        extensionScheme = 'ms-browser-extension';
-      } else if (/Chrome\//.test(userAgent)) {
-        extensionScheme = 'chrome-extension';
-      } else if (/Safari\//.test(userAgent)) {
-        extensionScheme = 'safari-extension';
-      } else {
-        extensionScheme = 'browserext';  // Upcoming standard scheme.
-      }
+    // IE does not support `document.currentScript` (nor extensions with protocol), so skip tests.
+    if (!msie) {
+      describe('auto bootstrap restrictions', function() {
 
-      var src = extensionScheme + '://something';
-      // Fake a minimal document object (the actual document.currentScript is readonly).
-      var fakeDoc = {
-        currentScript: { getAttribute: function() { return src; } },
-        location: {protocol: extensionScheme + ':', origin: extensionScheme + '://something'},
-        createElement: document.createElement.bind(document)
-      };
-      expect(allowAutoBootstrap(fakeDoc)).toBe(true);
+        function createFakeDoc(attrs, protocol, origin, currentScript) {
```

Comment:
4 parameters are not very readable. Could you get the protocol from the origin? `currentScript` & `attrs` could be unified, just check if `currentScript` is a plain object and create a true `script` that extends it if it is. Then you'd only have 2 params.

Otherwise a dictionary would be more readable (i.e. `function createFakeDoc({attrs, protocol, origin, currentScript})` but in ES5).